front/components/assistant_builder/avatar_picker/AssistantBuilderCustomUpload.tsx

```diff
@@ -36,50 +37,41 @@ const AssistantBuilderCustomUpload = React.forwardRef<
   const fileInputRef = useRef<HTMLInputElement>(null);
   const imageRef = useRef<HTMLImageElement | null>(null);
 
+  const fileUploaderService = useFileUploaderService({
+    owner,
+    useCase: "avatar",
+  });
+
   useImperativeHandle(ref, () => {
     return {
       getUrl: async () => {
         if (isUploadingAvatar) {
-          return;
+          return null;
         }
 
         if (imageRef.current && crop.width && crop.height) {
           const croppedImageUrl = await getCroppedImg(imageRef.current, crop);
-
-          const formData = new FormData();
           const response = await fetch(croppedImageUrl);
+
           const blob = await response.blob();
           const f = new File([blob], "avatar.jpeg", { type: "image/jpeg" });
-          formData.append("file", f);
 
           setIsUploadingAvatar(true);
+          const files = await fileUploaderService.handleFilesUpload([f]);
```

Comment:
@flvndvd see no evil

I can't do fileUploaderService.getFiles right after since the state is not yet ready :/

Reply:
Yeah here the issue is the `useImperativeHandle` hook. Let me take a deeper look.

Reply:
This component wasn't originally intended for this purpose, and the state requires additional ticks to process. It's acceptable for this PR, and I'll extract the API logic so it can be used outside of the component.